tests/src/gui/testqgsrelationreferencewidget.cpp

```diff
@@ -340,5 +341,38 @@ void TestQgsRelationReferenceWidget::testSetGetForeignKey()
   QCOMPARE( spy.count(), 3 );
 }
 
+
+// Test issue https://issues.qgis.org/issues/22071
+// Relation reference widget wrong feature when "on map identification"
+void TestQgsRelationReferenceWidget::testIdentifyOnMap()
+{
+  QWidget parentWidget;
+  QgsRelationReferenceWidget w( &parentWidget );
+  QVERIFY( mLayer1->startEditing() );
+  w.setRelation( *mRelation, true );
+  w.setAllowMapIdentification( true );
+  w.init();
+  QEventLoop loop;
+  // Populate model (I tried to listen to signals but the module reload() runs twice
+  // (the first load triggers a second one which does the population of the combo)
+  // and I haven't fin a way to properly wait for it.
```

Comment:
Maybe casting the combobox to a `QgsFeatureListComboBox` and then connecting to the models `isLoadingChanged` signal?
But it's not trivial

Reply:
> Maybe casting the combobox to a `QgsFeatureListComboBox` and then connecting to the models `isLoadingChanged` signal?
> But it's not trivial

I tried that, but the `isLoadingChanged` was triggered twice as well (4 times, really), and I missed the second `reload` (which actually loads the data in the combo), but I admit that the logic in that loading behavior is not clear to me. I thought about adding another signal but I'm not sure it would be useful out of the test, so I prefer to not add it.

Reply:
IIRC: The logic is to as quickly as possible get the display value of the current selected item to show up to the user. And then with a little delay lazy-load also some additional values to fill the popup.

Reply:
Maybe connecting to the signal and then checking if the rowCount of the model is > 1?